In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import urllib.request
import seaborn as sns
import scipy.stats as stats
import statsmodels.api as sm
import statsmodels.stats

In [2]:
# input data
feedback_raw = pd.read_csv('yelp_labelled.txt', delimiter= '\t', header=None)

# name columns
feedback_raw.columns=['text', 'target']

In [3]:
# tokenizer
from nltk.tokenize import word_tokenize


# split the data into training set and test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(feedback_raw['text'].values, 
                 feedback_raw['target'].values,        
                 test_size=0.2)

# vectorizer
from sklearn.feature_extraction.text import CountVectorizer
vect = CountVectorizer(tokenizer=word_tokenize)
tf_train = vect.fit_transform(X_train)
tf_test = vect.transform(X_test)




In [4]:
# logistic regression
from sklearn.linear_model import LogisticRegression
model = LogisticRegression(random_state=0, solver='lbfgs', multi_class='multinomial')
model.fit(tf_train, y_train)
preds = model.predict(tf_test)
accuracy = (preds == y_test).mean()
print('the accuracy of logistic regression is :{}'.format(accuracy))

the accuracy of logistic regression is :0.82


In [5]:
# multinomial naive bayes model
from sklearn.naive_bayes import MultinomialNB
modelMNB = MultinomialNB()
modelMNB.fit(tf_train, y_train)
preds = modelMNB.predict(tf_test)
accuracy = (preds == y_test).mean()
print('the accuracy of multinomial naive bayes is :{}'.format(accuracy))

the accuracy of multinomial naive bayes is :0.815


In [6]:
# use another file as the test 2
test_raw = pd.read_csv('amazon_cells_labelled.txt', delimiter= '\t', header=None)
test_raw.columns=['text', 'target']
X_test = test_raw['text'].values
y_test = test_raw['target'].values
tf_test = vect.transform(X_test)
preds = model.predict(tf_test)
accuracy = (preds == y_test).mean()
print('the accuracy for new data set using logistic regression is :{}'.format(accuracy))

the accuracy for new data set using logistic regression is :0.748


In [7]:
preds = modelMNB.predict(tf_test)
accuracy = (preds == y_test).mean()
print('the accuracy for new data set using multinomial naive bayes is :{}'.format(accuracy))

the accuracy for new data set using multinomial naive bayes is :0.738


Logistic Regression model and Multinomial Naive bayes regression model are used to the origianl dataset (yelp) and the new dataset (amazon).  In the yelp dataset, the accuracy is 0.82 while in amazon dataset, the accuracy is 0.74.  